dimod/constrained.py

```diff
@@ -773,9 +774,119 @@ def vartype(self, v: Variable) -> Vartype:
         """The vartype of the given variable."""
         return self.variables.vartype(v)
 
+    @classmethod
+    def from_lp_file(cls,
+                     fp: Union[BinaryIO, ByteString],
+                     lower_bound_default: Optional[int] = 0,
+                     upper_bound_default: Optional[int] = 1000) -> "ConstrainedQuadraticModel":
+        """
+        Create a CQM model from a LP file.
+        Args:
+            fp: file-like object in the LP format
+            lower_bound_default: in case lower bounds for integer are not given, this will be the defaults
+            upper_bound_default: in case upper bounds for integer are not given, this will be the defaults
 
-CQM = ConstrainedQuadraticModel
+        Returns:
+            The model encoded in the LP file, converted to ConstrainedQuadraticModel dimod object
+        """
+
+        return cqm_model_from_lp_file(fp, lower_bound_default, upper_bound_default)
 
 
+CQM = ConstrainedQuadraticModel
+
 # register fileview loader
 load.register(CQM_MAGIC_PREFIX, ConstrainedQuadraticModel.from_file)
+
+
+def cqm_model_from_lp_file(fp: Union[BinaryIO, ByteString],
```

Comment:
If this function is going to live here (which makes sense to avoid circular import), I would just move the contents into the method directly. No need for the separate function.